lib/ansible/module_utils/common/collections.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c), Sviatoslav Sydorenko <ssydoren@redhat.com> 2018
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+"""Collection of low-level utility functions."""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+try:
+    """Python 3.3+ branch."""
+    from collections.abc import (
+        deque, KeysView,
+        Mapping, MutableMapping,
+        Sequence, MutableSequence,
+        Set, MutableSet,
+    )
+except ImportError:
+    """Use old lib location under 2.6-3.2."""
+    from collections import (
+        deque, KeysView,
+        Mapping, MutableMapping,
+        Sequence, MutableSequence,
+        Set, MutableSet,
+    )
+
+
+from ansible.module_utils.six import binary_type, text_type
+
+
+def is_string_type(seq):
+    """Identify whether the input has a string-like type (inclding bytes)."""
+    return isinstance(seq, (text_type, binary_type))
+
+
+def is_iterable(seq):
+    """Identify whether the input is an iterable."""
+    try:
+        iter(seq)
+        return True
+    except TypeError:
+        return False
+
+
+def is_sequence(seq, include_strings=False):
+    """Identify whether the input is a sequence.
+
+    Strings and bytes are not sequences here,
+    unless ``include_string`` is ``True``.
+
+    Non-indexable things are never of a sequence type.
+    """
+    maybe_a_sequence = include_strings or not is_string_type(seq)
+    return maybe_a_sequence and isinstance(seq, Sequence)
```

Comment:
Please use if statements for readability instead of boolean logic.
``` python
# Slightly more readable.  Probably needs a comment to explain the string portion of the check:
# If include_strings is False, then we also have to return False if seq is a string.
if isinstance(seq, Sequence) and include_strings or not is_string_type(seq):
    return True
return False

# Probably most readable as two different conditional statements:
if not include_strings and is_string_type(seq):
    return False

return isinstance(seq, Sequence)
```

Reply:
I normally use variable naming for readability. I cannot agree that if-clauses are more readable in all cases, but I'll do as you ask.